h/static/scripts/tests/util/submit-form-test.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+var submitForm = require('../../util/submit-form');
+
+/**
+ * Fake `fetch()` implementation.
+ *
+ * This does not mock the global `window.fetch` function but provides a mock
+ * via `this.fetch` which can be passed as a dependency to functions that make
+ * network requests.
+ */
+class FakeFetch {
```

Comment:
Sinon already provides a full XHR stubbing toolkit, so let's not introduce assumptions into our tests about exactly *how* we utilise XHR, but instead stub requests/responses and ensure they result in the right behaviour.